ts/Series/SeriesOnPointComposition.ts

```diff
@@ -0,0 +1,563 @@
+            this.series = series;
+            this.options = series.options.onPoint;
+        }
+
+        /* *
+         *
+         *  Properties
+         *
+         * */
+
+        public chart: Chart;
+
+        public connector?: SVGElement;
+
+        public options?: OnPoint;
+
+        public radii?: Array<number>;
+
+        public series: SeriesComposition;
+
+        public zData?: zData;
+
+        /**
+         * @ignore
+         */
+        public getRadii = bubble.prototype.getRadii;
+
+        /**
+         * @ignore
+         */
+        public getRadius = bubble.prototype.getRadius;
+
+        /**
+         * @ignore
+         */
+        public getPxExtremes = bubble.prototype.getPxExtremes;
+
+        /**
+         * @ignore
+         */
+        public getZExtremes = bubble.prototype.getZExtremes;
+
+        /**
+         * Draw connector line that starts from the initial point's position
+         * and ends in the center of the series.
+         * @private
+         */
+        public drawConnector(): void {
+            if (!this.connector) {
+                this.connector = this.series.chart.renderer.path()
+                    .addClass('highcharts-connector-seriesonpoint')
+                    .attr({
+                        zIndex: -1
+                    })
+                    .add(this.series.markerGroup);
+            }
+
+            const attribs = this.getConnectorAttributes();
+
+            attribs && this.connector.attr(attribs);
+        }
+
+        /**
+         * Get connector line path and styles that connects series and point.
+         *
+         * @private
+         *
+         * @return {Highcharts.SVGAttributes} attribs - the path and styles.
+         */
+        public getConnectorAttributes(): SVGAttributes|void {
+            const chart = this.series.chart,
+                onPointOptions = this.options;
+
+            if (!onPointOptions) {
+                return;
+            }
+
+            const connectorOpts = onPointOptions.connectorOptions || {},
+                position = onPointOptions.position,
+                connectedPoint = chart.get(onPointOptions.id);
+
+            if (
+                !(connectedPoint instanceof Point) ||
+                !position ||
+                !defined(connectedPoint.plotX) ||
+                !defined(connectedPoint.plotY)
+            ) {
+                return;
+            }
+
+            const xFrom = defined(position.x) ?
+                    position.x :
+                    connectedPoint.plotX,
+                yFrom = defined(position.y) ?
+                    position.y :
+                    connectedPoint.plotY,
+                xTo = xFrom + (position.offsetX || 0),
+                yTo = yFrom + (position.offsetY || 0),
+                width = connectorOpts.width || 1,
+                color = (connectorOpts as any).color || this.series.color,
+                dashStyle = (connectorOpts as any).dashStyle;
```

Comment:
The any casts here are because `connectorOptions` is set as SVGAttributes type, but `color` and `dashStyle` don't exist on that type. Either we need a separate interface like `OnPointConnectorOptions`, or we need to use `stroke` and `dashstyle` instead of `color` and `dashStyle`. Either is fine for me.